pkg/controller/volume/persistentvolume/index.go

```diff
@@ -124,13 +126,23 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *v1.PersistentVol
 				if volumeQty.Cmp(requestedQty) < 0 {
 					continue
 				}
+				// filter out mismatching volumeModes
+				// TODO: remove feature gate check after no longer needed
+				if utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) {
```

Comment:
You could add the check at the beginning since it's common to both prebound and unbound PVs